File: forumClient/src/Dto/app/ChatInfo.ts
import type { UserBasicInfo } from "../UserInfo";
import type { ChatMessageBasicInfo, ChatMessageFullInfo } from "./ChatMessageInfo";

export type ChatEntry = {
    chatId: string;
    chatGroupId: string;
    chatName: string;
    createdByUserId: string;
    createdDate: string;
}

export type ChatBasicInfo = {
    chat: ChatEntry
    messages: ChatMessageBasicInfo[]
}

export type ChatFullInfo = {
    chat: ChatEntry,
    messages: ChatMessageFullInfo[],
    users: UserBasicInfo[]
}

File: server/RestApiServer/Dto/Chat/ChatInfo.cs
using RestApiServer.Db;

namespace RestApiServer.Dto.Chat
{
    public class ChatBasicInfo 
    {
        public ChatEntry Chat { get; set; } = null!;
    }
    public class ChatFullInfo
    {
        public ChatEntry Chat { get; set; } = null!;
        public List<ChatMessageBasicInfo> Messages { get; set; } = null!;
    }
}